src/pages/profiles/[id].tsx

```diff
@@ -39,9 +40,11 @@ import type {
         });
       },
     });
+
   
     if (profile == null || profile.name == null) {
-      return <ErrorPage statusCode={404} />;
+      return < LoadingSpinner />
+      // return <ErrorPage statusCode={404} />;
```

Comment:
The actual 404 pages aren't going to work now though. You'll need to separate the loading state and error state @ankush170